tests/test_hf_mpt_gen.py

```diff
@@ -72,3 +78,108 @@ def test_init_hfhub_mpt(device: str, attn_impl: str):
 
 def test_init_hfhub_mpt_cpu():
     test_init_hfhub_mpt(device='cpu', attn_impl='torch')
+
+
+@pytest.mark.gpu
+@patch.object(ComposerGenerate,
+              'generate',
+              side_effect=ComposerGenerate.generate,
+              autospec=True)
+def test_mpt_generate_callback(callback_generate: Any, tmpdir: Path):
+    composer_device = get_device('gpu')
+    reproducibility.seed_all(42)
+    max_seq_len = 128
+
+    # testing dataset and dataloader
+    dataset_size = 5
+
+    tiny_dataset_path = tmpdir / 'test-ift-data-small'
+    tiny_dataset_path.mkdir()
+    tiny_dataset_file = tiny_dataset_path / 'train.jsonl'
+    make_tiny_ft_dataset(path=str(tiny_dataset_file), size=dataset_size)
+
+    dataloader_cfg = {
+        'name': 'finetuning',
+        'dataset': {
+            'hf_name': str(tiny_dataset_path),
+            'split': 'train',
+            'max_seq_len': max_seq_len,
+            'decoder_only_format': True,
+            'allow_pad_trimming': False,
+            'packing_ratio': None,
+            'shuffle': True,
+        },
+        'drop_last': False,
+        'num_workers': 4,
+        'pin_memory': False,
+        'prefetch_factor': 2,
+        'persistent_workers': False,
+        'timeout': 0
+    }
+
+    dataloader_cfg = om.create(dataloader_cfg)
+
+    # build tokenizer
+    tokenizer_cfg: Dict[str, Any] = om.to_container(
+        DictConfig({
+            'name': 'EleutherAI/gpt-neox-20b',
+            'kwargs': {
+                'model_max_length': max_seq_len,
+            }
+        }),
+        resolve=True,
+    )  # type: ignore
+    tokenizer = build_tokenizer(tokenizer_cfg['name'],
+                                tokenizer_cfg.get('kwargs', {}))
+
+    # build mpt model
+    model_config = DictConfig({
+        'name': 'mpt_causal_lm',
+        'pretrained_model_name_or_path': 'mosaicml/mpt-7b',
+        'pretrained': False,
+        'config_overrides': {
+            'd_model': 128,
+            'n_heads': 4,
+            'n_layers': 2,
+            'expansion_ratio': 2,
+            'attn_config': {
+                'attn_impl': 'triton',
+            },
+        },
+    })
+    model = COMPOSER_MODEL_REGISTRY['mpt_causal_lm'](model_config, tokenizer)
+    model = composer_device.module_to_device(model)
+
+    # generate callback
+    prompts = [
+        'The best banana bread recipe is',
+        '2+2=',
+        'how much wood could a woodchuck chuck',
+    ]
+    gen_interval = 1
+    generate_cb = ComposerGenerate(
```

Comment:
Instead of patching generate in the function header, wdyt about wrapping ComposerGenerate in a Mock like so:
```
generate_cb = Mock(wraps=ComposerGenerate(...), autospec=True)

# other code...

#Assertion
generate_cb.generate.assert_called_once()
```